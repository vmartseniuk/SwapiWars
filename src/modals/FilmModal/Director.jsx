/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-one-expression-per-line */
import React, { memo } from "react";
import { makeStyles, Typography, Avatar } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  directorContainer: {
    display: "inline-flex",
  },
  directorIcon: {
    marginRight: 20,
  },
  directorInfo: {
    display: "flex",
    flexDirection: "column",
    alignItems: "flex-start",
    justifyContent: "center",
  },
  directorLabel: {
    ...theme.typography.tiny,
    color: "#726A6A",
    textTransform: "capitalize",
  },
}));

const Director = (props) => {
  const { children, src } = props;
  const classes = useStyles();

  return (
    <div className={classes.directorContainer}>
      <Avatar alt="Profile Picture" src={src} className={classes.directorIcon} />
      <div className={classes.directorInfo}>
        <Typography className={classes.directorName}>{children}</Typography>
        <Typography className={classes.directorLabel}>producer</Typography>
      </div>
    </div>
  );
};

export default memo(Director);
