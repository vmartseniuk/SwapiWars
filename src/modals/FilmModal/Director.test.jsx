import { create } from "react-test-renderer";
import Director from "./Director";

describe("FilmModal/Director", () => {
  it("renders correctly", () => {
    const rendered = create(<Director>test name</Director>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
