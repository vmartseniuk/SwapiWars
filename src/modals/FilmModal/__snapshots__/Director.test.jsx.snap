// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmModal/Director renders correctly 1`] = `
<div
  className="makeStyles-directorContainer-1"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle makeStyles-directorIcon-2 MuiAvatar-colorDefault"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiAvatar-fallback"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <div
    className="makeStyles-directorInfo-3"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      test name
    </p>
    <p
      className="MuiTypography-root makeStyles-directorLabel-4 MuiTypography-body1"
    >
      producer
    </p>
  </div>
</div>
`;
